core/commands/dag/dag.go

```diff
@@ -241,3 +247,100 @@ var DagResolveCmd = &cmds.Command{
 	},
 	Type: ResolveOutput{},
 }
+
+var DagExportCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Streams the selected DAG as a .car stream on stdout.",
+		ShortDescription: `
+'ipfs dag export' fetches a dag and streams it out as a well-formed .car file.
+Note that at present only single root selections / .car files are supported.
+The output of blocks happens in strict DAG-traversal, first-seen, order.
+`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("root", true, false, "CID of a root to recursively export").EnableStdin(),
+	},
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+
+		c, err := cid.Decode(req.Arguments[0])
+		if err != nil {
+			return fmt.Errorf(
+				"unable to parse root specification (currently only bare CIDs are supported): %s",
+				err,
+			)
+		}
+
+		node, err := cmdenv.GetNode(env)
+		if err != nil {
+			return err
+		}
+
+		// Code disabled until descent-issue in go-ipld-prime is fixed
+		//
+		// The second part of the above - make a super-thin wrapper around
+		// a blockservice session, translating Session.GetBlock() to Blockstore.Get()
+		//
+		// The current interface of go-car is rather suboptimal as it
+		// only takes a blockstore, instead of accepting a dagservice,
+		// and leveraging parallel-fetch capabilities
+		// https://github.com/ipld/go-car/issues/27
+		//
+		//
+		// sess := blockservice.NewSession(
+		// 	req.Context,
+		// 	node.Blocks,
+		// )
+		// var wrapper getBlockFromSessionWrapper = func(c cid.Cid) (blk.Block, error) {
+		// 	return sess.GetBlock(req.Context, c)
+		// }
+		// sb := gipselectorbuilder.NewSelectorSpecBuilder(gipfree.NodeBuilder())
+		// car := gocar.NewSelectiveCar(
+		// 	req.Context,
+		// 	&wrapper,
+		// 	[]gocar.Dag{gocar.Dag{
+		// 		Root: c,
+		// 		Selector: sb.ExploreRecursive(
+		// 			gipselector.RecursionLimitNone(),
+		// 			sb.ExploreAll(sb.ExploreRecursiveEdge()),
+		// 		).Node(),
+		// 	}},
+		// )
+
+		pipeR, pipeW := io.Pipe()
+
+		errCh := make(chan error, 2) // we only report the 1st error
+		go func() {
+			defer func() {
+				if err := pipeW.Close(); err != nil {
+					errCh <- fmt.Errorf("stream flush failed: %s", err)
+				}
+				close(errCh)
+			}()
+
+			//if err := car.Write(pipeW); err != nil {
+			if err := gocar.WriteCar(
+				req.Context,
+				node.DAG,
+				[]cid.Cid{c},
+				pipeW,
+			); err != nil {
+				errCh <- err
+			}
+		}()
+
+		if err := res.Emit(pipeR); err != nil {
+			return err
+		}
+
+		err = <-errCh
+
+		// minimal user friendliness
+		if err != nil &&
+			!node.IsOnline &&
+			err.Error() == "merkledag: not found" {
```

Comment:
`err == format.ErrNotFound`

@hsanjuan is also working on improving this error message.